docker/ubuntu-based/prod/vswitch/vppctl

```diff
@@ -4,6 +4,14 @@
 
 set -euo pipefail
 
+get_vswitch_container_ID() {
+    # There may be more than one vswitch container if an upgrade/redeploy is
+    # happening, so go for the newest one.
+    CID=$(docker ps --format '{{.CreatedAt}} {{.Image}} {{.ID}}' | grep -E " $1@?[^ ]* [^ ]+$" | sort -nr | \
+        head -n 1 | sed -e 's|.* ||g') || [ -z "$CID" ]
```

Comment:
The:
|| [ -z "$CID" ] 
Shouldn't be here.  Additionally, there is no point setting a local variable then echoing it.  Just do:
docker ps --format '{{.CreatedAt}} {{.Image}} {{.ID}}' | grep -E " $1@?[^ ]* [^ ]+$" | sort -nr | \
    head -n 1 | sed -e 's|.* ||g'

Reply:
This is just repeating what the script was doing before. If the '|| [ -z "$CID" ]' is not there, and there is no container present, the script will fail. If i don't put '|| [ -z "$CID" ]' here, i need to put it in all places where this function is called (as you point out later). Is that what you prefer?